src/test/TxTests.cpp

```diff
@@ -196,48 +202,29 @@ applyCheck(TransactionFramePtr tx, Application& app, bool checkSeqNum)
             tx->getResult().result.code(txINTERNAL_ERROR);
         }
         REQUIRE((!res || tx->getResultCode() == txSUCCESS));
-
-        if (!check)
+        if (applyResult)
```

Comment:
I don't think this is what we want as this still leaves room for unchecked behavior:
the reason we have this function is to ensure that `checkValid` is consistent with `apply` (while having sane default checks that allow us to not have test code too bloated).

Right now the code still allows for conditions where we ignore the value of `checkResult` (`applyResult` set but in the case `!check` we don't check for anything if `validationResult` is not set).

Basically: if either `validationResult` or `applyResult` are not set, we have to have very strict checks in place where we have 100% confidence in the correctness of the results (so the default checks passing means we have a non exceptional behavior).

Reply:
Now it is not possible to set only `validationResult` or `applyResult`, so that should be solved.